### PR TITLE
Feat/486 missing mdu keywords coastal

### LINKED ISSUE #486: Missing keywords for (3D) coastal models
**Describe the bug**
For 3D models like DCSM-FM we deviate from some default settings. To do so we make use of several keywords that are not available in FMModel of hydrolib-core yet:

**Update 5 April 2023:**
In Stendert's list below I (@arthurvd) have listed if and how we will support this (awaiting approval by some colleagues):
 :heavy_check_mark: : in User Manual, support it
 :mortar_board: : Research keyword in 6th gen models, support it
 :grey_question: : undecided
:x: : no support in hydrolib-core

in _[numerics]_:
- Keepstbndonoutflow  :mortar_board:
- Baroczlaybed :mortar_board:  *Extra added by developer*
- Barocponbnd :mortar_board:
- MinTimestepBreak :heavy_check_mark:

in _[physics]_:
- ITcap :mortar_board:
- Tempmax :mortar_board:
- Tempmin :mortar_board:
- Salimax :mortar_board:
- Salimin :mortar_board:
- Heat_eachstep :mortar_board:
- RhoairRhowater :mortar_board:
- Nudgetimeuni :mortar_board: *Extra added by developer*
- IniWithNudge :mortar_board:

in _[time]_:
- AutoTimestep :mortar_board:
- Startdatetime :mortar_board:
- Stopdatetime :mortar_board:

in _[trachytopes]_:
- TrtMxR :heavy_check_mark:

in _[output]_:
- FouFile :heavy_check_mark:
- fouUpdateStep :mortar_board: *Extra added by developer*
- WaqInterval :heavy_check_mark: *was in hydrolib-core already*
- StatsInterval :heavy_check_mark: *was in hydrolib-core already*
- Wrihis_turbulence :heavy_check_mark:
- Wrihis_wind :heavy_check_mark:
- Wrihis_rain :heavy_check_mark:
- Wrihis_infiltration :heavy_check_mark:  *Extra added by developer*
- Wrihis_temperature :heavy_check_mark:  *Extra added by developer*
- Wrihis_waves :heavy_check_mark:
- Wrihis_heat_fluxes :heavy_check_mark:
- Wrihis_salinity :heavy_check_mark:  *Extra added by developer*
- Wrihis_density :heavy_check_mark:
- Wrihis_waterlevel_s1 :heavy_check_mark:  *Extra added by developer*
- Wrihis_bedlevel :heavy_check_mark:  *Extra added by developer*
- Wrihis_waterdepth :heavy_check_mark:  *Extra added by developer*
- Wrihis_velocity_vector :heavy_check_mark:  *Extra added by developer*
- Wrihis_upward_velocity_component :heavy_check_mark:  *Extra added by developer*
- Wrihis_sediment :heavy_check_mark:
- Wrihis_constituents :heavy_check_mark:
- Wrihis_zcor :heavy_check_mark:  *Extra added by developer*
- Wrihis_taucurrent :heavy_check_mark:
- Wrimap_evaporation :heavy_check_mark:
- Wrimap_waterdepth :heavy_check_mark:
- Wrimap_velocity_magnitude :heavy_check_mark:
- Writek_CdWind :x: *Tekplot no official support/not maintained*
- Wrimap_CdWind :x: *does not exist!*
- Wrimap_tidal_potential :heavy_check_mark:
- Wrimap_sal_potential :heavy_check_mark: *Extra added by developer*
- Wrimap_internal_tides_dissipation :heavy_check_mark: *Extra added by developer*
- Richardsononoutput :heavy_check_mark: *was in hydrolib-core already*

**To Reproduce**
1. Run the lines:
```
from pathlib import Path
import hydrolib.core.dflowfm as hcdfm

oldPath = r'p:\11208054-004-dcsm-fm\models\3D_DCSM-FM\2013-2017\B05\DCSM-FM_0_5nm.mdu'
mduOld  = hcdfm.FMModel(filepath=oldPath,path_style='unix',recurse=False)

newPath = r'p:\11208054-004-dcsm-fm\models\3D_DCSM-FM\2013-2017\B05_hydrolib\DCSM-FM_0_5nm.mdu'
mduOld.filepath=newPath
mduOld.save(path_style='unix',recurse=False)
```
2. Start a short simulation (done): p:\11208054-004-dcsm-fm\models\3D_DCSM-FM\2013-2017\B05_hydrolib\
3. Compare the diagnostics output to see which keywords in the used mdu changed:
- p:\11208054-004-dcsm-fm\models\3D_DCSM-FM\2013-2017\B05\DFM_OUTPUT_DCSM-FM_0_5nm\DCSM-FM_0_5nm_0001.dia
- p:\11208054-004-dcsm-fm\models\3D_DCSM-FM\2013-2017\B05_hydrolib\DFM_OUTPUT_DCSM-FM_0_5nm\DCSM-FM_0_5nm_0001.dia

**Expected behavior**
After reading and writing an mdu there should be no differences in settings (and in the resulting simulation).

**Version info (please complete the following information):**
 - OS: Windows 10

### PR DESCRIPTION
**Pull request by arthurvd**
Fixes #486

### CODE REVIEW

**Comment by veenstrajelmer** [comment]
Tested with the workflow from the corresponding issue, there are quite some differences in the diafiles:
image

File: hydrolib/core/dflowfm/mdu/models.py [hunk 1]
```diff
@@ -815,6 +896,13 @@ class Comments(INIBasedModel.Comments):
             "Space separated list of files, containing information about observation cross sections.",
             alias="crsFile",
         )
+        foufile: Optional[str] = Field(
+            "Fourier analysis input file *.fou", alias="fouFile"
```
Comment (priscavdsluis) [inline hydrolib/core/dflowfm/mdu/models.py:900]: Isn't it an output file?

File: hydrolib/core/dflowfm/mdu/models.py [hunk 2]
```diff
@@ -1024,6 +1174,18 @@ class Comments(INIBasedModel.Comments):
...
+            alias="wrimap_tidal_potential",
+        )
+        wrimap_sal_potential: Optional[str] = Field(
+            "Write self attraction and loading potential to map file (1: yes, 0: no)",
+            alias="wrimap_SAL_potential",
+        )
+        wrimap_internal_tides_dissipation: Optional[str] = Field(
+            "Write internal tides dissipation to map file (1: yes, 0: no)",
+            alias="wrimap_internal_tides_dissipation",
+        )
```
Comment (priscavdsluis) [inline hydrolib/core/dflowfm/mdu/models.py:1188]: I am missing the keyword `wrimap_waterdepth` and `Wrimap_velocity_magnitude `

**Review by priscavdsluis — CHANGES_REQUESTED** [review]
@arthurvd Some questions and comments

File: hydrolib/core/dflowfm/ini/util.py [hunk 3]
```diff
@@ -266,6 +267,37 @@ def validate_conditionally(
...
+    Returns:
+        Optional[str]: the original input value, if valid.
+
+    Raises:
+        ValueError: if a non-empty input string does not have valid format.
+    """
+    if (
+        field_value is not None
+        and len(field_value.strip()) > 0
+        and field_value != "yyyymmddhhmmss"
```
Comment (priscavdsluis) [inline hydrolib/core/dflowfm/ini/util.py:289]: Is this expected input?

File: hydrolib/core/dflowfm/mdu/models.py [hunk 4]
```diff
@@ -128,6 +131,10 @@ class Comments(INIBasedModel.Comments):
         tlfsmo: Optional[str] = Field(
             "Fourier smoothing time on water level boundaries [s].", alias="tlfSmo"
         )
+        keepstbndonoutflow: Optional[str] = Field(
+            "Keep sal and tem signals on bnd also at outflow, 1=yes, 0=no=default=copy inside value on outflow",
```
[the remark this answers is not in the document]
Comment (arthurvd) [reply hydrolib/core/dflowfm/mdu/models.py:135]: fixed

File: hydrolib/core/dflowfm/mdu/models.py [hunk 5]
```diff
@@ -370,6 +396,24 @@ class Comments(INIBasedModel.Comments):
             "Coefficient for evaporative heat flux ( ), if negative, then Cd wind is used.",
             alias="dalton",
         )
+        tempmax: Optional[str] = Field("Limit the temperature", alias="tempMax")
+        tempmin: Optional[str] = Field("Limit the temperature", alias="tempMin")
+        salimax: Optional[str] = Field("Limit the salinity", alias="saliMax")
+        salimin: Optional[str] = Field("Limit the salinity", alias="saliMin")
```
[the remark this answers is not in the document]
Comment (arthurvd) [reply hydrolib/core/dflowfm/mdu/models.py:402]: Done in newer commit.
About optionality: I now made them non-optional consistently. We need to work under different issue on optional keywords and them not appearing in output MDU, see: #495.

File: hydrolib/core/dflowfm/mdu/models.py [hunk 6]
```diff
@@ -415,6 +460,14 @@ class Comments(INIBasedModel.Comments):
...
     stanton: float = Field(0.0013, alias="stanton")
     dalton: float = Field(0.0013, alias="dalton")
+    tempmax: float = Field(-999.0, alias="tempMax")
+    tempmin: float = Field(0.0, alias="tempMin")
+    salimax: float = Field(-999.0, alias="saliMax")
+    salimin: float = Field(0.0, alias="saliMin")
+    heat_eachstep: bool = Field(False, alias="heat_eachStep")
+    rhoairrhowater: int = Field(0, alias="rhoAirRhoWater")
+    nudgetimeuni: float = Field(3600.0, alias="nudgeTimeUni")
+    iniwithnudge: int = Field(0, alias="iniWithNudge")
```
[the remark this answers is not in the document]
Comment (arthurvd) [reply hydrolib/core/dflowfm/mdu/models.py:470]: I prefer to to this in a future improvement issue: there is currently many non-Enum ints in mdu.models. OK? See: #523.

File: hydrolib/core/dflowfm/mdu/models.py (hunk 1 above)
Comment (arthurvd) [reply hydrolib/core/dflowfm/mdu/models.py:900]: No, it's the Fourier configuration input file. The output file is `*_fou.nc`.

File: hydrolib/core/dflowfm/mdu/models.py (hunk 2 above)
Comment (arthurvd) [reply hydrolib/core/dflowfm/mdu/models.py:1188]: Added

**Review by priscavdsluis — APPROVED** [review]
Review OK!
Removed a code smell

**Comment by veenstrajelmer** [comment]
Tested and approved

**Pull request merged**